src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -689,6 +689,11 @@ public static bool CanRemoveParentheses(this ParenthesizedPatternSyntax node)
             if (pattern is ParenthesizedPatternSyntax)
                 return true;
 
+            // We're parenthesized discard pattern. We cannot remove parens.
+            // x is (_)
+            if (pattern is DiscardPatternSyntax { Parent: ParenthesizedPatternSyntax { Parent: IsPatternExpressionSyntax } })
```

Comment:
Isn't the `Parent` always ParenthesizedPatternSyntax since the `pattern` is coming from a `ParenthesizedPatternSyntax.Node`? This could potentially be simplified to:

```suggestion
            if (pattern is DiscardPatternSyntax { Parent: { Parent: IsPatternExpressionSyntax } })
```

Reply:
It can't be after this because `GetOperatorPrecedence` for discard pattern returns `OperatorPrecedence.Primary`

```cs
var patternPrecedence = pattern.GetOperatorPrecedence();
if (patternPrecedence == OperatorPrecedence.Primary || patternPrecedence == OperatorPrecedence.Unary)
    return true;
```